lib/puppet/type/ini_setting.rb

```diff
@@ -120,7 +120,7 @@ def insync?(current)
          'Defaults to undef (autodetect).'
   end
 
-  newparam(:refreshonly) do
+  newparam(:refreshonly, boolean: true, parent: Puppet::Parameter::Boolean) do
```

Comment:
Could you drop the `newvalues(true, false)` line below, as the boolean parent parameter will set that up correctly.

Reply:
Is it okay to use Ruby 2 keyword arguments above? Or does it still need to support ruby 1.9?

Reply:
Other than that question, I verified refreshonly behaves as expected with old and new puppet versions.

Reply:
we don't care about Ruby 1.9

Reply:
What about when puppetserver running jruby 1.7 (which only supports MRI 1.9 language features) loads the type to compile a catalog?

Reply:
To the point @joshcooper is making here, it looks like this change (and similar changes where functions like `newfunction(:create_ini_settings, :type => :statement, :doc => <<-EOS` have been updated as `newfunction(:create_ini_settings, type: :statement, doc: <<-EOS` at https://github.com/puppetlabs/puppetlabs-inifile/compare/2.0.0...2.2.0#diff-6b9f1e8fde3f18501b209c68609a51b0L6) are responsible for some pipeline breakages.

The agent upgrade module is now experiencing Ruby parser / compile failures executing tests, for instance in the cell `PLATFORM=redhat5-64default.a-redhat7-64mdca` at https://jenkins-master-prod-1.delivery.puppetlabs.net/view/All/job/forge-module_puppetlabs-puppet-agent-module_intn-sys_smoke-master/PLATFORM=redhat5-64default.a-redhat7-64mdca,WORKER_LABEL=beaker/33/consoleFull against the 2.2.0 release of this module, where it previously passed against the 2.0.0 release of this module.

I'm still parsing logs to make sure my understanding is accurate, but the messages being emitted seem to point strongly in this direction:

```
13:41:40      Beaker::Host::CommandFailure:
13:41:40        Host 'gzzk5kgvpmnkqfe.delivery.puppetlabs.net' exited with 1 running:
13:41:40         puppet apply --verbose --detailed-exitcodes /tmp/apply_manifest.pp.ox7060
13:41:40        Last 10 lines of output were:
13:41:40        	                          ^
13:41:40        	/etc/puppet/modules/inifile/lib/puppet/type/ini_setting.rb:134: syntax error, unexpected kEND, expecting $end
13:41:40        	Error: Evaluation Error: Error while evaluating a Function Call, Could not autoload puppet/type/ini_setting: /etc/puppet/modules/inifile/lib/puppet/type/ini_setting.rb:22: syntax error, unexpected ':', expecting ')'
13:41:40        	  newparam(:name, namevar: true) do
13:41:40        	                          ^
13:41:40        	/etc/puppet/modules/inifile/lib/puppet/type/ini_setting.rb:134: syntax error, unexpected kEND, expecting $end at /etc/puppet/modules/puppet_agent/manifests/prepare/puppet_config.pp:48:14 on node gzzk5kgvpmnkqfe.delivery.puppetlabs.net
13:41:40        	Error: Evaluation Error: Error while evaluating a Function Call, Could not autoload puppet/type/ini_setting: /etc/puppet/modules/inifile/lib/puppet/type/ini_setting.rb:22: syntax error, unexpected ':', expecting ')'
13:41:40        	  newparam(:name, namevar: true) do
13:41:40        	                          ^
13:41:40        	/etc/puppet/modules/inifile/lib/puppet/type/ini_setting.rb:134: syntax error, unexpected kEND, expecting $end at /etc/puppet/modules/puppet_agent/manifests/prepare/puppet_config.pp:48:14 on node gzzk5kgvpmnkqfe.delivery.puppetlabs.net
13:41:40        
13:41:40      # ./.bundle/gems/gems/beaker-3.32.0/lib/beaker/host.rb:375:in `exec'
13:41:40      # ./.bundle/gems/gems/beaker-3.32.0/lib/beaker/dsl/helpers/host_helpers.rb:83:in `block in on'
13:41:40      # ./.bundle/gems/gems/beaker-3.32.0/lib/beaker/shared/host_manager.rb:127:in `run_block_on'
13:41:40      # ./.bundle/gems/gems/beaker-3.32.0/lib/beaker/dsl/patterns.rb:37:in `block_on'
13:41:40      # ./.bundle/gems/gems/beaker-3.32.0/lib/beaker/dsl/helpers/host_helpers.rb:63:in `on'
13:41:40      # ./.bundle/gems/gems/beaker-puppet-0.10.0/lib/beaker-puppet/helpers/puppet_helpers.rb:505:in `block in apply_manifest_on'
13:41:40      # ./.bundle/gems/gems/beaker-3.32.0/lib/beaker/shared/host_manager.rb:127:in `run_block_on'
13:41:40      # ./.bundle/gems/gems/beaker-3.32.0/lib/beaker/dsl/patterns.rb:37:in `block_on'
13:41:40      # ./.bundle/gems/gems/beaker-puppet-0.10.0/lib/beaker-puppet/helpers/puppet_helpers.rb:434:in `apply_manifest_on'
13:41:40      # ./.bundle/gems/gems/beaker-puppet-0.10.0/lib/beaker-puppet/helpers/puppet_helpers.rb:512:in `apply_manifest'
13:41:40      # ./spec/acceptance/class_spec.rb:57:in `block (5 levels) in <top (required)>'
```

Note that the above error is emitted during a `puppet apply` on the Redhat 5 agent.